### PR TITLE
[CI] Updated drone.star file to match latest changes

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This is an update on drone.star file 
- moving docker image references to constants
- using caches images

## Related Issue
Part of: 
- https://github.com/owncloud/QA/issues/729
- https://github.com/owncloud/QA/issues/728

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -3,13 +3,41 @@
...
 OC_CI_GOLANG = "owncloudci/golang:1.17"
-OC_CI_NODEJS = "owncloudci/nodejs:14"
-OC_CI_PHP = "owncloudci/php:7.4"
+OC_CI_PHP = "owncloudci/php:%s"
 OC_CI_WAIT_FOR = "owncloudci/wait-for:latest"
-OC_TESTING_MIDDLEWARE = "owncloud/owncloud-test-middleware:1.3.1"
+OC_OC_TEST_MIDDLEWARE = "owncloud/owncloud-test-middleware:1.3.1"
 MINIO_MC = "minio/mc:RELEASE.2021-10-07T04-19-58Z"
 REDIS = "redis:6-alpine"
+BANST_AWS_CLI = "banst/awscli"
```
Comment (wkloucek) [inline .drone.star:13]: that one looks like it's not used?

File: .drone.star [hunk 2]
```diff
@@ -3,13 +3,41 @@
...
+PLUGINS_GH_PAGES = "plugins/gh-pages:1"
+PLUGINS_DOWNSTREAM = "plugins/downstream:latest"
+SELENIUM_STANDALONE_CHROME_DEBUG = "selenium/standalone-chrome-debug:3.141.59"
+OC_CI_DRONE_ANSIBLE = "owncloudci/drone-ansible:latest"
+OC_CI_BAZEL_BUILDIFIER = "owncloudci/bazel-buildifier:latest"
+MELTWATER_DRONE_CACHE = "meltwater/drone-cache:v1"
+OC_SERVER = "owncloud/server:10"
+OC_UBUNTU = "owncloud/ubuntu:18.04"
+MARIADB = "mariadb:10.6"
+OSIXIA_OPEN_LDAP = "osixia/openldap:latest"
```
[the remark this answers is not in the document]
Comment (pascalwengerter) [reply .drone.star:37]: I suppose that alphabetical is expected/easiest to process? :)

[the remark this answers is not in the document]
Comment (pascalwengerter) [reply .drone.star:37]: Ah sry, thought this was the `web` codebase. Dunno what the oCIS team prefers

[the remark this answers is not in the document]
Comment (wkloucek) [reply .drone.star:37]: alphabetical order sounds sane (and every IDE should have automatic sorting function)

[the remark this answers is not in the document]
Comment (phil-davis) [reply .drone.star]: sorted - my IDE had a "Sort lines" menu option.

File: .drone.star (hunk 1 above)
Comment (phil-davis) [reply .drone.star]: removed

**Pull request merged**